tests/boards/test_base.py

```diff
@@ -169,46 +159,35 @@ def test_board_group_boards_zero():
 
     assert len(board_group.boards) == 0
 
-    with pytest.raises(IndexError):
-        board_group.boards[0]
-
-
-def test_board_group_board_by_int():
-    """Test that the boards property works with int indices."""
-    board_group = BoardGroup(MockBoard, OneBoardMockBackend())
-
-    assert type(board_group[0]) == MockBoard
+    with pytest.raises(KeyError):
+        board_group.boards["SERIAL0"]
 
 
 def test_board_group_board_by_serial():
     """Test that the boards property works with serial indices."""
     board_group = BoardGroup(MockBoard, OneBoardMockBackend())
+    BoardGroup(MockBoard, OneBoardMockBackend())
 
-    assert type(board_group["SERIAL"]) == MockBoard
+    assert type(board_group[list(board_group.boards.values())[0].serial]) == MockBoard
 
 
 def test_board_group_board_by_unknown():
     """Test that the boards property throws an exception with unknown indices."""
     board_group = BoardGroup(MockBoard, OneBoardMockBackend())
 
+    with pytest.raises(KeyError):
+        board_group[0]
+
     with pytest.raises(KeyError):
         board_group[""]
 
-    with pytest.raises(IndexError):
+    with pytest.raises(KeyError):
```

Comment:
If you change the above comment from `KeyError` to `TypeError`, I think you need to change it here too.
```suggestion
    with pytest.raises(TypeError):
```